vlm/src/test/scala/geotrellis/contrib/vlm/gdal/GDALRasterSourceSpec.scala

```diff
@@ -75,4 +81,66 @@ class GDALRasterSourceSpec extends FunSpec with RasterMatchers with BetterRaster
       assertRastersEqual(actual, expected)
     }
   }
+
+  describe("should perform a tileToLayout") {
+    val cellSizes = {
+      val tiff = GeoTiffReader.readMultiband(url)
+      (tiff +: tiff.overviews).map(_.rasterExtent.cellSize).map { case CellSize(w, h) =>
+        CellSize(w + 1, h + 1)
+      }
+    }
+
+    cellSizes.foreach { targetCellSize =>
+      it(s"should perform a tileToLayout for cellSize: ${targetCellSize}") {
+        val pe = ProjectedExtent(source.extent, source.crs)
+        val scheme = FloatingLayoutScheme(256)
+        val layout = scheme.levelFor(pe.extent, targetCellSize).layout
+        val mapTransform = layout.mapTransform
+        val resampledSource = source.resampleToGrid(layout)
+
+        val expected: List[(SpatialKey, MultibandTile)] =
+          mapTransform(pe.extent).coordsIter.map { spatialComponent =>
+            val key = pe.translate(spatialComponent)
+            val ext = mapTransform.keyToExtent(key.getComponent[SpatialKey])
+            val raster = resampledSource.read(ext).get
+            val newTile = raster.tile.prototype(source.cellType, layout.tileCols, layout.tileRows)
+            key -> newTile.merge(
+              ext,
+              raster.extent,
+              raster.tile,
+              NearestNeighbor
+            )
+          }.toList
+
+        val layoutSource = source.tileToLayout(layout)
+        val actual: List[(SpatialKey, MultibandTile)] = layoutSource.readAll().toList
+
+        withClue(s"actual.size: ${actual.size} expected.size: ${expected.size}") {
+          actual.size should be(expected.size)
+        }
+
+        val sortedActual: List[Raster[MultibandTile]] =
+          actual
+            .sortBy { case (k, _) => (k.col, k.row) }
+            .map { case (k, v) => Raster(v, mapTransform.keyToExtent(k)) }
+
+        val sortedExpected: List[Raster[MultibandTile]] =
+          expected
+            .sortBy { case (k, _) => (k.col, k.row) }
+            .map { case (k, v) => Raster(v, mapTransform.keyToExtent(k)) }
+
+        val grouped: List[(Raster[MultibandTile], Raster[MultibandTile])] =
+          sortedActual.zip(sortedExpected)
+
+        grouped.foreach { case (actualTile, expectedTile) =>
+          withGeoTiffClue(actualTile, expectedTile, source.crs) {
+            assertRastersEqual(actualTile, expectedTile)
+          }
+        }
+
+        resampledSource.close
+        layoutSource.source.close
```

Comment:
@jbouffard @echeipesh the first time i actually had to clean up memory, without it i had:

```
error] Error during tests:
[error] 	Running java with options -classpath ... <the huge classpath there> ...
```

Reply:
Was there a useful message with this error or just that ?